src/NUnitTestExtractor/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -183,5 +202,67 @@ public static Level ParseLevel(string levelToParse)
             }
             return level;
         }
+
+        /// <summary>
+        /// Appends all of a TestCase's arguments to a string
+        /// </summary>
+        /// <param name="args">array of arguments to append to string</param>
+        /// <param name="data">the string to add to and also return</param>
+        /// <returns>all of the arguments in a string with proper formatting</returns>
+        public static string AppendArgumentsForTestCasesToString(object[] args, string data)
+        {
+            if(args != null)
+            {
+                int count = args.Length;
+
+                for (int i = 0; i < args.Length; i++)
+                {
+                    var arg = args[i];
+
+                    string argString = null;
+
+                    if (arg != null)
+                        argString = arg as string;
+
+                    //if its not last(dont need comma) and if the arg after it is not null(also dont need a comma)
+                    if (i != count - 1)
```

Comment:
You can probably simplify this if/else a bit by creating a `List<string> testArgs` and just adding the various args to that list.  Then when you've added all the arguments, you can join them with commas with `string.Join(", ", testArgs)`